test/sampling_correlation_matrices_test.cpp

```diff
@@ -36,6 +36,21 @@ MT rebuildMatrix(const VT &xvector, const unsigned int n){
     return mat;
 }
 
+template<typename NT, typename MT>
+Eigen::Matrix<NT, Eigen::Dynamic, 1> getCoefficientsFromMatrix(const MT& mat) {
```

Comment:
why do you need this function? Notice that the class ` CorreMatrix` includes this representation. Look at the member function `getCoefficients()`.

However, it's not clear why this function is needed. Can you comment on this?

Reply:
here in the uniform_correlation_sampling_MT() function

```
for(const auto&p : randPoints){
        MT final_cor_mat = p.mat + p.mat.transpose() - MT::Identity(n, n);
    	randCorMatrices.push_back(final_cor_mat);
    }
```

p.mat, etc. return type MT, so I'm actually storing MT and not CorreMatrix<NT> in `std::list<MT> randCorMatrices`. 

but getCoefficients() is a member function of the CorreMatrix class, so in the testing function in the loop

```
for(const auto& mat : randCorMatrices){
    	    samples.col(jj) = getCoefficientsFromMatrix<NT, MT>(mat);
    	    jj++;
	}
```

since every member of randCorMatrices is MT, I couldn't call the getCoefficients() function which is why I wrote the separate function to get the coefficients

I thought we need this loop for populating the samples variable

Reply:
ic, then let's keep it in the test folder